app/jobs/concerns/import_job.rb

```diff
@@ -63,42 +53,15 @@ def category_model
     @category_model ||= category.constantize
   end
 
-  def stats
-    @stats ||= Hash.new(0)
-    @stats[:shl_case_numbers] = Hash.new(0) unless @stats.key?(:shl_case_numbers)
-    @stats
-  end
-
-  def increment_stats(attrs, persisted)
-    stats[:row_count] += 1
-    if persisted
-      stats[:rows_imported] += 1
-      stats[:shl_case_numbers][attrs['shl_case_number']] += 1 if attrs.key?('shl_case_number')
-    else
-      stats[:rows_not_imported] += 1
-    end
-  end
-
-  # We're receiving date values in a format different than what ActiveRecord expects, we need a mechanism
-  # for applying transformations to values before they get to the model.
-  def preprocess_attribute_values(attrs)
-    attrs
-  end
-
-  def translate_attribute_names(attrs)
-    attrs
-  end
-
   def already_processed?(filename)
     ProcessedFile.where(status: 'Processed').where(original_filename: original_filename(filename)).count > 0
   end
 
   def initialize_processed_file(filename)
-    @processed_file = ProcessedFile.create(filename: filename,
-                                           original_filename: original_filename(filename),
-                                           category: category,
-                                           status: 'Running',
-                                           job_stats: stats)
+    @processed_file = ProcessedFile.new(filename: filename,
+                                        original_filename: original_filename(filename),
+                                        category: category,
+                                        status: 'Running')
```

Comment:
Only instantiate the `ProcessedFile` when initializing. The save occurs in the complete/fail_processed_file! methods after stats from the import are collected.